packages/devtools_app/lib/src/screens/deep_link_validation/deep_links_model.dart

```diff
@@ -163,13 +168,122 @@ class IosDomainError extends DomainError {
     'Ensure your domain is accessible without any redirects.',
   );
 
-  /// TODO(hangyujin): There are sub checkes of this check, add them and add links when finalized.
-  /// AASA file format follows guidelines.
-  static const fileFormat = IosDomainError(
-    'Apple-App-Site-Association file format is incorrect',
-    'This test checks that your Apple-App-Site-Association file '
-        'follows the correct format guidelines.',
-    'Ensure your Apple-App-Site-Association file follows the correct format guidelines.',
+  static IosDomainError iosFileFormatDomainError({
+    required List<AASAfileFormatSubCheck> subcheckErrors,
+  }) =>
+      IosDomainError(
+        'Apple-App-Site-Association file format is incorrect',
+        'This test checks that your Apple-App-Site-Association file '
+            'follows the correct format guidelines.',
+        'Ensure your Apple-App-Site-Association file follows the correct format guidelines.',
+        subcheckErrors: subcheckErrors,
+      );
+
+  final List<AASAfileFormatSubCheck> subcheckErrors;
+}
+
+class AASAfileFormatSubCheck extends CommonError {
+  const AASAfileFormatSubCheck(title, explanation)
+      : super(title, explanation, '');
+
+  static const appLinksFormat = AASAfileFormatSubCheck(
+    'Applinks format',
+    'This test checks if the `applinks` property holds an object.',
```

Comment:
nit: in all the error explanations above, we use the verbiage "checks that" instead of "checks if". Let's use "checks that" where applicable in these errors for consistency